### PR TITLE
Ajax code

### PR DESCRIPTION
**Pull request by 70ray**
Let ajax function return all data on not ok
so we can examine the error code if necessary.
Add a function ajaxAlert() to alert with the error message.

### CODE REVIEW

File: scripts/api.js [hunk 1]
```diff
@@ -25,20 +25,25 @@ function ajax(method, apiUrl, queryParams = {}, data = {}, fetchPromise = fetch)
             .then(function(response) {
                 const contentType = response.headers.get('content-type');
                 if (!contentType || !contentType.includes('application/json')) {
-                    reject("Incorrect response type");
+                    reject({error: "Incorrect response type", code: 998});
```
[the remark this answers is not in the document]
Comment (70ray) [reply scripts/api.js]: Yes, I'll do this.

**Review by cpeel — APPROVED** [review]
Returning the full error block to the client makes sense. I'm a little wary of the convention of having the client `alert()` the error to the user. That means we need to design error messages for the end user, not the client developer.

**Comment by cpeel** [comment]
Note: we should squash these commits on merge.

**Review by srjfoo — APPROVED** [review]
Bearing in mind that I know absolutely nothing about ajax, and that the code means nothing to me, I tested the page browser with invalid, nonexistent and archived projects, and with a nonexistent page number, comparing TEST with the branch, and didn't see any discrepancies.

**Review by chrismiceli — APPROVED** [review]

File: scripts/api.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 /*global codeUrl */
-/* exported ajax */
+/* exported ajax ajaxAlert unknownError incorrectResponseType networkError */
+
+const unknownError = 999;
```
[the remark this answers is not in the document]
Comment (70ray) [reply scripts/api.js]: Done now.

**Review by chrismiceli — APPROVED** [review]

**Comment by 70ray** [comment]
Sandbox updated with latest changes.

**Pull request merged**